packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -82,6 +82,7 @@ class KeySet<T extends KeyboardKey> {
   final HashSet<T> _keys;
 
   @override
+  // ignore: avoid_equals_and_hash_code_on_mutable_classes
```

Comment:
This one seems odd. Why is this one allowed to be mutable, but still have an overridden operator==? @gspencergoog 

We should add a comment why this ignore is legit (see style guide).

Reply:
@yjbanov rewrote this part recently to cache the hash code, but other than that, it's possible it could be modified to use a factory method and be const. It would mean a significant performance loss though (up to 9x worse).

This seems like a legit ignore because the reason it is mutable is so that the hash code can be cached: the reason for the exception is directly related to storing the computed hash code, not the things that go into computing it.

Reply:
Yeah, that seems like a good solution for this.